src/main/scala/mesosphere/marathon/core/launchqueue/impl/TaskLauncherActor.scala

```diff
@@ -205,6 +213,25 @@ private class TaskLauncherActor(
   }
 
   private[this] def receiveTaskLaunchNotification: Receive = {
+    case InstanceOpSourceDelegate.InstanceOpRejected(op, TaskLauncherActor.OfferOperationRejectedTimeoutReason) =>
+      // Reschedule instance with provision timeout..
+      if (inFlightInstanceOperations.exists(_.instanceId == op.instanceId)) {
+        instanceMap.get(op.instanceId).foreach { instance =>
+          import scala.concurrent.ExecutionContext.Implicits.global
+
+          logger.info(s"Reschedule ${instance.instanceId} because of provision timeout.")
+          async {
+            if (runSpec.isResident) {
+              await(instanceTracker.process(RescheduleReserved(instance, runSpec.version)))
```

Comment:
We have an interesting case here. This command overrides the state of the tasks to `Scheduled`. Three things can happen now:

1. The provisioned tasks were reported running but the `TaskLauncherActor` did not know. We falsely reschedule the tasks.
2. The provisioned tasks never become running and Mesos sends and offer with the reservations again.
3. The provisioned tasks are reported running *after* we rescheduled them.

Cases 2 and 3 are fine. Since the tasks will be running.
Case 1 is interesting. The tasks are running but Marathon is waiting for an offer with the reservation. That will never come. However, after a reconciliation a Mesos update will override the tasks condition to running again.

Reply:
I am kind of afraid it will be race-y and at the same time we don't have any integration test covering this so ... not sure :)

I mean you reschedule here, in the meantime mesos update might come with that task id, also you reschedule, but you still have in instanceMap the old instance, until you get InstanceChange event and sync that... 

Ideally I would like to see some integration tests loop but since we don't have tests for that I guess we have to believe? :) and fix this soon via decommission? (can you please create JIRA for this? so that we don't forget)

Reply:
> I mean you reschedule here, in the meantime mesos update might come with that task id, also you reschedule, but you still have in instanceMap the old instance, until you get InstanceChange event and sync that...

It's actually not that racy as it will converge. Even when the test comes back as failed. We might overscale ephemeral tasks but the scale check will take care of these cases.

> Ideally I would like to see some integration tests loop but since we don't have tests for that I guess we have to believe? :) and fix this soon via decommission? (can you please create JIRA for this? so that we don't forget)

We don't kill with a `decommission` but you are right. Once we have that we should reschedule in a different way. Ideally through incarnation counts. See [MARATHON-8373](https://jira.mesosphere.com/browse/MARATHON-8373).

I''l set up a loop but doubt that it would show anything since the timeout is high enough.

Reply:
I switched our [loop](https://jenkins.mesosphere.com/service/jenkins/view/Marathon/job/marathon-sandbox/job/marathon-loop-goals-phase-2/) to this branch.